### ISSUE TITLE
The use of probe loss in ssl setting

### ISSUE THREAD

**Issue by arghadeep2001** [rbalestr-lab/lejepa#39]
Excellent work!
But i have a question related to the loss function.
In traditional ssl setting we always train the model in a self suervised manner(whithout any labeled data) ; then use backbone embedding as a feature extractor and train a prediction head(related to downstream task) on the top of that.

But here loss function looks like this
                lejepa_loss = sigreg_loss * cfg.lamb + inv_loss * (1 - cfg.lamb)
                y_rep, yhat = y.repeat_interleave(cfg.V), probe(emb.detach())
                probe_loss = F.cross_entropy(yhat, y_rep)
                loss = lejepa_loss + probe_loss

In probe_loss calculation we need the labelled data. 
So are we going to train the whole ssl+downstreamtask as a single loss function?

**Comment by RandallBalestriero** [comment]
Thank you! So in the example code we used an `online probe` to produce metrics during SSL training. *The very important thing is to detach the embedding before feeding it to the probe* otherwise this is just supervised learning. In a real-world scenario where you do not have labels at training time, you would just remove that probe and use lejepa_loss alone. Hope this helps!
